jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -1164,14 +1164,15 @@ private boolean lockedIsWriting()
         private Runnable lockedFailWrite(Throwable x)
         {
             assert _request._lock.isHeldByCurrentThread();
-            Callback writeCallback = _writeCallback;
-            _writeCallback = null;
-            if (writeCallback == null)
-                return null;
+            // We always record the failure here, so even if there is no write, subsequent writes will fail.
             if (_writeFailure == null)
                 _writeFailure = x;
             else
                 ExceptionUtil.addSuppressedIfNotAssociated(_writeFailure, x);
+            Callback writeCallback = _writeCallback;
```

Comment:
The 4 lines above could be replaced by `_writeFailure = ExceptionUtil.combine(x)`